contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/rest/problem/ContentGridExceptionHandler.java

```diff
@@ -0,0 +1,111 @@
+package com.contentgrid.spring.data.rest.problem;
+
+import com.contentgrid.spring.data.rest.problem.ext.ConstraintViolationProblemProperties;
+import com.contentgrid.spring.data.rest.problem.ext.ConstraintViolationProblemProperties.FieldViolationProblemProperties;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonMappingException.Reference;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.hibernate.exception.ConstraintViolationException;
+import org.springframework.context.support.MessageSourceAccessor;
+import org.springframework.data.rest.core.RepositoryConstraintViolationException;
+import org.springframework.hateoas.mediatype.problem.Problem;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+@ResponseBody
+@RequiredArgsConstructor
+public class ContentGridExceptionHandler {
+
+    @NonNull
+    private final ProblemFactory problemFactory;
+
+    @NonNull
+    private final MessageSourceAccessor messageSourceAccessor;
+
+    @NonNull
+    private final JsonPropertyPathConverter jsonPropertyPathConverter;
+
+    @ExceptionHandler(ConstraintViolationException.class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    Problem handleConstraintViolation(ConstraintViolationException exception) {
+        var type = switch (exception.getSQLState()) {
+            case "23505" -> ProblemType.UNIQUE_CONSTRAINT_VIOLATION;
+            default -> ProblemType.CONSTRAINT_VIOLATION;
+        };
+        return problemFactory.createProblem(type, exception.getConstraintName())
+                .withStatus(HttpStatus.CONFLICT);
```

Comment:
I don't think Conflict is appropriate for constraint violations that _aren't_ a unique constraint violation.

Reply:
I agree. However, I'm not sure what status to assign to which class exactly.

From [the Postgres docs](https://www.postgresql.org/docs/current/errcodes-appendix.html)


|Class 23| Integrity Constraint Violation | HTTP status code |
| ----------| ------------ | ---------- |
| 23000 | integrity_constraint_violation | :question: |
| 23001 | restrict_violation | :question: |
| 23502 | not_null_violation | 400 Bad Request |
| 23503 | foreign_key_violation | :question: |
| 23505 | unique_violation | 409 Conflict | 
| 23514 | check_violation | :question: |
| 23P01 | exclusion_violation | 409 Conflict |

Reply:
In case of doubt: plain HTTP 400